github/github.go

```diff
@@ -424,6 +424,33 @@ func (c *Client) NewRequest(method, urlStr string, body interface{}) (*http.Requ
 	return req, nil
 }
 
+// NewFormRequest creates an API request. A relative URL can be provided in urlStr,
+// in which case it is resolved relative to the BaseURL of the Client.
+// Relative URLs should always be specified without a preceding slash.
+// Body is sent with Content-Type: application/x-www-form-urlencoded.
+func (c *Client) NewFormRequest(urlStr string, body io.Reader) (*http.Request, error) {
+	if !strings.HasSuffix(c.BaseURL.Path, "/") {
+		return nil, fmt.Errorf("BaseURL must have a trailing slash, but %q does not", c.BaseURL)
+	}
+
+	u, err := c.BaseURL.Parse(urlStr)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequest(http.MethodPost, u.String(), body)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
It would be nice if this case got tested.

Reply:
There seems to be no case where this `http.NewRequest` returns an error.

There are two cases in which `http.NewRequest` returns an error: when the method is invalid or when the url fails to parse. `NewFormRequest` is set to POST, so the former error does not occur. Also, since url parsing is done in `c.BaseURL.Parse` before this process, if url parsing fails, an error is returned before this process.

Even the already existing NewUploadRequest does not seem to cover this case. Is there a policy for such cases?

Reply:
Could you pass in an `iotest.ErrReader` as the body to `NewRequest` to make it fail?
https://pkg.go.dev/testing/iotest#ErrReader

Reply:
thanks. It didn't work. Just generating a `http.Request` will not cause it to be read by the `iotest.ErrReader`.

Reply:
OK, thanks for trying.  I'm OK with that.
Merging.